### PR TITLE
Feature: Delete container UI in group single page

### PR DESCRIPTION
**Pull request by ShoeheyOt**
## Overview

Implement the delete container dialog

## Changes

- move delete button `DropdownMenuButton` to different component named `DeleteContainerDialogTriggerButton` this is same structure as Group dropdown menu.

## Review points

- as I talked with @nick-y-ito , we don't need 
```tsx
const handleCancel = () => {
    closeParentOnCancel && onParentClose?.();
  };
```
for deleteing Group/Container, so I deleted this from `DeleteGroupDialogContent.tsx`. I would like you to review if it is OK, I checked by myself and it is correctly working, but to make sure.

- I changed some code in some file which is not target scope of this PR, please ignore that part, I left some comment ,which you can ignore

## Screen Captures

[DeleteContainer_UI.webm](https://github.com/nishiki-tech/nishiki-frontend/assets/142355969/e4b3dd40-47be-454a-8aae-93987f58e902)

## Assignee Checklist:

- [X] The naming convention of the PR title is correct (See the comment at the top of this template)
- [X] The base branch is correct (See: [Types of Branches](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [X] The branch name follows the [Branch Naming Conventions](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [X] The correct assignees and reviewers have been designated for this PR
- [X] The coding style follows the [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [X] All the related issues are associated with this PR
- [X] All criteria in the associated issues are met (please tick the checkboxes)
- [X] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

- [x] Double-check the "Assignee Checklist"
- [x] The code follows the generic best practices and our [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] The code is well-commented and easy to understand
- [x] The UI changes accurately reflect the provided design

### CODE REVIEW

File: src/features/groups/components/ContainerCardList/ContainerCard/ContainerCardDropdownMenuTriggerButton/ContainerCardDropdownMenuContent/DeleteContainerDialogTriggerButton/DeleteContainerDialogContent.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+   */
+  onDialogClose: () => void;
+}
+
+export const DeleteContainerDialogContent = ({
+  onParentClose,
+  onDialogClose,
+}: IDeleteContainerDialogContentProps) => {
+  /**
+   * The function to handle when delete button clicked
```
Comment (nick-y-ito) [inline src/features/groups/components/ContainerCardList/ContainerCard/ContainerCardDropdownMenuTriggerButton/ContainerCardDropdownMenuContent/DeleteContainerDialogTriggerButton/DeleteContainerDialogContent.tsx:27]: Please describe in the TSDoc that this function closes both the parent and this dialog.
To ensure we remember it when you add functionality in the next PR.

**Review by nick-y-ito — CHANGES_REQUESTED** [review]
@ShoeheyOt 
Thank you for your work! Its behavior looks having no problems! Please take a look at the comments I left, are mainly for how to add comments.

File: src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/GroupCardDropdownMenuTriggerButton/GroupCardDropdownMenuContent/DeleteGroupDialogTriggerButton/DeleteGroupDialogContent.tsx
[the remark this answers is not in the document]
Comment (ShoeheyOt) [reply src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/GroupCardDropdownMenuTriggerButton/GroupCardDropdownMenuContent/DeleteGroupDialogTriggerButton/DeleteGroupDialogContent.tsx:1]: @nick-y-ito Thank you for the review!
Let me confirm with you that your comment here is, I should leave the function itself and modify the content of it like below?

```tsx
 /**
   * The function to close dropdown menu when cancel button clicked
   */
  const handleCancel = () => {
    onParentClose();
  }
```

[the remark this answers is not in the document]
Comment (nick-y-ito) [reply src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/GroupCardDropdownMenuTriggerButton/GroupCardDropdownMenuContent/DeleteGroupDialogTriggerButton/DeleteGroupDialogContent.tsx:1]: Exactly! Here are more specific versions:

```tsx
  /**
   * Handle the cancel button click.
   * It closes the parent UI component if specified.
   */

  /**
   * Event handler triggered on cancel button click.
   * It closes the parent UI component if specified.
   */
```

`onParentClose` should be an optional prop even if you delete `closeParentOnCancel`.
```tsx
  const handleCancel = () => {
    onParentClose?.();
  }
```

[the remark this answers is not in the document]
Comment (nick-y-ito) [reply src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/GroupCardDropdownMenuTriggerButton/GroupCardDropdownMenuContent/DeleteGroupDialogTriggerButton/DeleteGroupDialogContent.tsx:1]: > It closes the parent UI component together with this dialog

I liked the revised comment thought. You choose :)

**Review by nick-y-ito — APPROVED** [review]
@ShoeheyOt 
LGTM! Thank you for all of your work!

**Pull request merged**